Pull request mvc-framework

Zavodsk0y opened the pull request:
Done. Я "провафлил" прошлый репоз (я каким-то образом закинул на первоначальных этапах проект в мейн), потому пришлось организовать данный новый репозиторий с этой веткой :\

Я до конца так и не разобрался, что делать с композером, ибо как я понял, он composer.json видит только в main-ветке без поддиректорий, потому у меня не получается нормально залить пакет на packagist.org

Существующие зависимости можно глянуть в composer.json, если уж на то пошло

БДшку в админере мне пришлось создавать самостоятельно, ибо он выдавал, что ее не существует

Управление БД осуществляется по адресу:

localhost/adminer

?В .env вносятся такие же значения параметров БД, что указаны в pop-it-mvc/config/db.php?

Zavodsk0y committed:
Done. Я "провафлил" прошлый репоз (я каким-то образом закинул на первоначальных этапах проект в мейн), потому пришлось организовать данный новый репозиторий с этой веткой :\

jurapro commented on README.md:
```diff
@@ -1 +1,117 @@
+    
+5. Выполните команду для установки зависимостей:
+
+    composer require illuminate/database illuminate/events
```
А почему так, если эта зависимость уже в composer.json?

jurapro commented on README.md:
```diff
@@ -1 +1,117 @@
+
+Существующие зависимости можно глянуть в composer.json, если уж на то пошло
+
+БДшку в админере мне пришлось создавать самостоятельно, ибо он выдавал, что ее не существует
```
Странно, в docker-compose.yml обычно указываешь БД и норм, он ее создает если нету...

jurapro commented on pop-it-mvc/app/Model/User.php:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    //Выборка пользователя по первичному ключу
+    public function findIdentity(int $id)
```
возвращаемый тип

jurapro commented on pop-it-mvc/app/Model/User.php:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    //Возврат аутентифицированного пользователя
+    public function attemptIdentity(array $credentials)
```
возвращаемый тип

jurapro commented on pop-it-mvc/core/Src/Auth/Auth.php:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    //Возврат текущего аутентифицированного пользователя
+    public static function user()
```
тип

jurapro commented on pop-it-mvc/core/Src/Request.php:
```diff
@@ -0,0 +1,50 @@
+        return $this->body + $this->files();
+    }
+
+    public function set($field, $value): void
```
можно field типизировать

jurapro commented on pop-it-mvc/core/Src/Request.php:
```diff
@@ -0,0 +1,50 @@
+        $this->body[$field] = $value;
+    }
+
+    public function get($field)
```
можно field типизировать

jurapro commented on pop-it-mvc/core/Src/View.php:
```diff
@@ -0,0 +1,79 @@
+        global $app;
+        $root = $app->settings->getRootPath();
+        $path = $app->settings->getViewsPath();
+        return '/app/pop-it-mvc' . $root . $path; // document_root не срабатывает из-за корневой папки nginx
```
можно было сделать document_root/.../ - чтобы он на уровень выше поднялся

jurapro replied:
или в чем-то другом проблема?

jurapro reviewed (changes requested):
- [ ] исправить замечания
- [ ] добавить решение тестового задания с помощью этого фреймворка (функционал гостя)

Zavodsk0y committed:
Типизация

Zavodsk0y committed:
Начинаем в апи. Делаем немного костыльно. Обновляем таблицу пользователей, добавляя поле под токен + генерируем его при регистрации/авторизации

Авторизация пока нерабочая.

Просмотр списка товаров по своей идее = просмотру постов

Zavodsk0y committed:
Доделываем api. Реализуем рабочую регистрацию и авторизацию через password_hash вместо md5.
Добавляем новый валидатор PasswordLengthValidator, который будет обязывать пользователя вводить пароль длиной свыше 6 символов
Поле для токена в таблице с пользователями заведено, при логине токен постоянно обновляется.

jurapro reviewed (approved):
принято
